laurent22/joplin
# Server: Resolves #9355: Add sync debug page to Joplin Server

# Summary

Adds a new `/sync_debug` route to Joplin Server that provides the following information:
- whether an item is on the server
- whether an item shows up in the initial `diff`

To take advantage of existing rate-limiting code, information is fetched from a client-side script rather than on the server.

Resolves #9355.

# Notes

Joplin Server doesn't seem to have a bundler or other workflow set up for converting TypeScript to client-side JavaScript. As such, most of the new code is in **JavaScript, rather than TypeScript.** 

# Screenshot

![screenshot of the debug page](https://github.com/laurent22/joplin/assets/46334387/7f977841-6371-4b53-8db5-0d71aef4c763)

# Testing plan

1. Start Joplin Server in development mode
2. Sync a client to Joplin Server
3. Copy the markdown link to a note
4. Open `http://localhost:22300/sync_debug` in a browser
5. Paste the markdown link into the `Check if an item is present on the server` input box and press `Submit`
6. Paste the markdown link into the `Check if an item is in the diff used for an initial sync` input box and press `Submit`
7. Repeat for a resource with a file URL, the external link to a notebook, and a note uuid
8. Repeat with a UUID that does not correspond to an item
